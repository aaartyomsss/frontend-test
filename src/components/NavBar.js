import React from 'react';
import { Button, makeStyles, Grid } from '@material-ui/core';
import { NavLink } from 'react-router-dom';

const useStyles = makeStyles({
    root: {
        margin: "auto",
        width: '100%'
    },
    grid: {
        textAlign: 'center'
    }
})

const styles = {
    border: '1px solid black',
    backgroundColor: '#222831'
}

const link = {
    textDecoration: 'none',
    width: '100%',
    color: '#ececec'
}

const NavBar = () => {

    const classes = useStyles();

    return (
        <div style={styles}>
            <Grid container>
                <Grid item xs={6} className={classes.grid}>
                    <Button className={classes.root}>
                        <NavLink style={link} to='/users'>Users</NavLink>
                    </Button>
                </Grid>
                <Grid item xs={6} className={classes.grid}>
                    <Button className={classes.root}>
                        <NavLink style={link} to='/projects'>Projects</NavLink>
                    </Button>
                </Grid>
            </Grid>
        </div>
    )

}

export default NavBar;