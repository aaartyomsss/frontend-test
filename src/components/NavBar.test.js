import React from 'react'
import '@testing-library/jest-dom/extend-expect'
import { render } from '@testing-library/react'
import { shallow, mount } from 'enzyme'
import NavBar from './NavBar'
import { BrowserRouter as Router } from 'react-router-dom'

describe('NavBar component', () => {

    test('Component renders corectly', () => {
        const component = shallow(<NavBar/>)
        expect(component).toBeDefined()
    })


    test('Component has 2 buttons', () => {
        const component = render(<Router><NavBar/></Router>)
        expect(component.container.querySelectorAll('button')).toHaveLength(2)
    })

    test('Button is clickable', () => {
        const component = mount(<Router><NavBar/></Router>)
        component.find('button#navBarTest').simulate('click')
        expect(component).toMatchSnapshot()
        component.unmount()
    })
})