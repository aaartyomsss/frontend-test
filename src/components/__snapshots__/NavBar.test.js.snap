// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component Button is clickable 1`] = `ReactWrapper {}`;
